src/lib.rs

```diff
@@ -298,6 +324,126 @@ mod contract {
         // TODO: https://github.com/aurora-is-near/aurora-engine/issues/2
     }
 
+    #[no_mangle]
+    pub extern "C" fn new_eth_connector() {
+        EthConnectorContract::init_contract()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn deposit_near() {
+        EthConnectorContract::new().deposit_near()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn withdraw_near() {
+        EthConnectorContract::new().withdraw_near()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn deposit_eth() {
+        EthConnectorContract::new().deposit_eth()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn withdraw_eth() {
+        EthConnectorContract::new().withdraw_eth()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn finish_deposit_near() {
+        EthConnectorContract::new().finish_deposit_near();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn finish_deposit_eth() {
+        EthConnectorContract::new().finish_deposit_eth();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_total_supply() {
+        EthConnectorContract::new().ft_total_supply();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_total_supply_near() {
+        EthConnectorContract::new().ft_total_supply_near();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_total_supply_eth() {
+        EthConnectorContract::new().ft_total_supply_eth();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_balance_of() {
+        EthConnectorContract::new().ft_balance_of();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_balance_of_eth() {
+        EthConnectorContract::new().ft_balance_of_eth();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_transfer() {
+        EthConnectorContract::new().ft_transfer();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn transfer_near() {
+        EthConnectorContract::new().transfer_near();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn transfer_eth() {
+        EthConnectorContract::new().transfer_eth();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_resolve_transfer() {
+        EthConnectorContract::new().ft_resolve_transfer();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn ft_transfer_call() {
+        EthConnectorContract::new().ft_transfer_call();
+    }
+
+    #[no_mangle]
+    pub extern "C" fn storage_deposit() {
+        EthConnectorContract::new().storage_deposit()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn storage_withdraw() {
+        EthConnectorContract::new().storage_withdraw()
+    }
+
+    #[no_mangle]
+    pub extern "C" fn storage_balance_of() {
+        EthConnectorContract::new().storage_balance_of()
+    }
+
+    #[cfg(feature = "integration-test")]
```

Comment:
The main question is - how?

Reply:
Ideally the prover contract stub would be a separate resource only used in the tests and it would be deployed to the simulated environment as part of the `init` setup in the tests.

The part I'm not sure on is if we would need to commit the compiled wasm for that stub to the repo, or if it could be compiled as part of the test. I know nearcore has some contracts that are only used for testing, so we can look at what was done there.

Reply:
Thanks for the idea! I think we will implement it as the next step of testing.